### PR TITLE
SEG-7 Add chat widget to Hubspot destination

### PR DESCRIPTION
**Pull request by pankaj-webonise**
### Ticket:

https://webonise.atlassian.net/browse/SEG-7

### CODE REVIEW

File: integrations/hubspot/test/index.test.js [hunk 1]
```diff
@@ -53,6 +55,12 @@ describe('HubSpot', function() {
         analytics.page();
         analytics.assert(window._hsq instanceof Array);
       });
+
+      it('should check for loadChatWidget to open chat widget', function() {
+        if (this.options.loadChatWidget) {
+          window.HubSpotConversations.widget.load();
```
Comment (ameyawebonise) [inline integrations/hubspot/test/index.test.js]: where is the assertion?

Comment (pankaj-webonise) [reply integrations/hubspot/test/index.test.js]: Updated the test. and one more thing I found that we are checking for the method call.

File: integrations/hubspot/test/index.test.js [hunk 2]
```diff
@@ -37,13 +38,15 @@ describe('HubSpot', function() {
...
         .option('loadFormsSdk', false)
+        .option('loadChatWidget', false)
         .option('portalId', null)
     );
   });
 
   describe('before loading', function() {
     beforeEach(function() {
       analytics.stub(hubspot, 'load');
+      analytics.stub(window.HubSpotConversations.widget, 'load');
```
Comment (ameyawebonise) [inline integrations/hubspot/test/index.test.js]: why do you want to change the behaviour for each test case?Why not limit it to the current test case that you are writing?

Comment (pankaj-webonise) [reply integrations/hubspot/test/index.test.js]: I found that every stub is created in beforeEach(). that's why I did this.

Comment (ameyawebonise) [reply integrations/hubspot/test/index.test.js]: try to see if you can have a before step in our test case.That way the stub will be limited to only our test run

**Review by ameyawebonise — APPROVED** [review]

**Review by brennan — APPROVED** [review]

**Comment by pankaj-webonise** [comment]
@brennan 
When I am adding code in index.js it is throwing an error while running tests. can you please help me with it?

Error:
1) should check for loadChatWidget to open chat widget
     HubSpot before loading #initialize
     TypeError: Cannot read property 'widget' of undefined
    at Context.<anonymous> (/tmp/test/index.test.js:60 <- /tmp/58d6b4e74120e3195772042cb525ec0b.browserify.js:1023:52)

2) should load the analytics lib and the forms lib if that option is true
     HubSpot loading
     Error: Timeout of 2000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves.

**Comment by brennan** [comment]
Hey @pankaj-webonise, could you push the code you've implemented to this branch so I can take a look? Thanks.

**Review by brennan — CHANGES_REQUESTED** [review]
It looks as if `window.HubSpotConversations` is not available on the page, hence the test failure. Before we can continue with this update, someone on Segment's end will have to figure out what upgrades are needed to pull in the correct version of Hubspot's JS.

**Review by CarlosMecha — COMMENTED** [review]
We can close this PR. Turns out the customers currently can add the widget if they create an inbox in their HubSpot dashboard, without any modification in our end.

**Pull request closed without merging**